integration_tests/test_domestic_migration.py

```diff
@@ -63,14 +174,74 @@ def test_individuals_move_into_new_households(simulants_on_adjacent_timesteps):
+        observed_numerator=all_time_joining_recently_created_households.sum(),
+        observed_denominator=len(all_time_joining_recently_created_households),
+        # This case -- people who joined a just-created household -- should be exceedingly rare.
+        target_proportion=(
+            (1 / 1_000_000_000),
+            0.001,
+        ),
+        name_additional="All time steps",
+    )
+    fuzzy_checker.fuzzy_assert_proportion(
+        "Domestic migration creating new households",
+        observed_numerator=all_time_new_household_movers.sum(),
+        observed_denominator=len(all_time_new_household_movers),
+        target_proportion=multiplicative_drift_to_bounds_through_timestep(
+            target_migration_rates["individual"]["new_household"],
+            target_migration_rates["multiplicative_drift"]["lower_bound"],
+            target_migration_rates["multiplicative_drift"]["upper_bound"],
+            len(simulants_on_adjacent_timesteps),
+        ),
+        name_additional="All time steps",
+    )
 
-def test_individuals_move_into_group_quarters(simulants_on_adjacent_timesteps):
-    for before, after in simulants_on_adjacent_timesteps:
+
+def test_individuals_move_into_group_quarters(
+    simulants_on_adjacent_timesteps, fuzzy_checker: FuzzyChecker, target_migration_rates
+):
+    all_time_gq_movers = []
+    all_time_into_gq_from_gq = []
+
+    for time_step, (before, after) in enumerate(simulants_on_adjacent_timesteps):
         gq_movers = (before["household_id"] != after["household_id"]) & (
             after["household_details.housing_type"] != "Household"
         )
-        assert gq_movers.any()
-        assert (before[gq_movers]["household_details.housing_type"] == "Household").any()
+        fuzzy_checker.fuzzy_assert_proportion(
+            "Domestic migration into GQ",
+            observed_numerator=gq_movers.sum(),
+            observed_denominator=len(gq_movers),
+            target_proportion=multiplicative_drift_to_bounds_at_timestep(
+                target_migration_rates["individual"]["gq_person"],
+                target_migration_rates["multiplicative_drift"]["lower_bound"],
+                target_migration_rates["multiplicative_drift"]["upper_bound"],
+                time_step,
+            ),
+            name_additional=f"Time step {time_step}",
+        )
+        all_time_gq_movers.append(gq_movers)
+        into_gq_from_gq = before[gq_movers]["household_details.housing_type"] != "Household"
+        fuzzy_checker.fuzzy_assert_proportion(
+            "Domestic migration into GQ from GQ",
+            observed_numerator=into_gq_from_gq.sum(),
+            observed_denominator=len(into_gq_from_gq),
+            # Since this is sampled randomly from the population only based on demographics, it should
+            # be around 3%, with substantial leeway for demographics correlations
+            target_proportion=(
+                data_values.PROP_POPULATION_IN_GQ * 0.5,
```

Comment:
Note: This is an unusual one where I think setting the target UI on the engineering side makes sense. It can be very conservative. It's kind of a strange assertion (see line 73 on the deleted side).

Reply:
I think this is in the same category as the one above.